### ISSUE TITLE
Folder Browser - Better look and feel on smaller screens

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/hpi#655]
See the screenshots below.  The folder browser should look better in 720p resolution:

Full screen:
![image](https://cloud.githubusercontent.com/assets/1377228/7283218/c708f11a-e8fa-11e4-8e4f-5057514afaf5.png)

Folder browser - folder names start getting cut off even one level deep:
![image](https://cloud.githubusercontent.com/assets/1377228/7283229/d8b207ee-e8fa-11e4-9095-045b20fed01c.png)

Folder browser - Going two levels deep causes both horizontal and vertical scrollbars:
![image](https://cloud.githubusercontent.com/assets/1377228/7283250/f6e8cb30-e8fa-11e4-8bf1-307c48a691e8.png)

Doc Viewer - even with a handful of actions, they span three lines:
![image](https://cloud.githubusercontent.com/assets/1377228/7283285/1f3af77a-e8fb-11e4-938b-b234538e62a5.png)

**Comment by gsteimer** [comment]
Some ideas:

**Folder Browser Wrapping**

1. This one should be simple - just slide the whole thing a bit over to the left.  There's definitely some wasted whitespace on the left
1. I know we've talked about this one, but... 
  * When the folder tree gets too wide, dynamically add width to the folder browser and remove width from the folder results pane

**Small Doc Viewer**
The goal of these sections are:
* Folder results pane - see all the documents in the folder
* Doc viewer - view the document, execute actions
* Give the user the ability to quickly view other documents in the folder

Give the above, what if we do this when a user clicks to view a document:
* Show the doc in the viewer on the right
* Hide the folder browser
* Make the doc viewer take up 80% or so of the screen
* Remove the folder contents pane and replace it with a new (smaller) pane:
  * List the docs in the folder, but with *only* the link attribute.  For most clients, this would simply be a list of the object names
    * I guess this could be a new pane, or maybe there's a way we can dynamically hide the other columns in the table and then shrink down the existing pane?  The trick with this option would be to remember which columns the user had visible when expanding the table again (see below).
  * If a user clicks another document link, simply put that doc in the viewer
  * Potentially add the "launch in right pane" icon, like the stage.  With this view layout, we could eventually have side by side viewing in the folder browser.  We don't have to do this now unless it's easy.
* If the user clicks the [X] on the right to close the doc viewer, go back to the layout with the folder browser and doc results table.

@joehof @cmlewis @aviking - thoughts?

**Comment by joehof** [comment]
For the folder browser wrapping, I'm of the opinion of not just adding width, but just make the whole thing scroll. Names won't wrap at all, they stay as they are and the user scrolls right to see them.

I think giving the docviewer 80% of screen when shown is good. I think we should pretty easily be able to make folder contents re-render only with the link attr, we would basically need to strip all the other attrs off of the config and then feed the smaller config to the view. Obviously we would hold onto the original config for when we were just viewing folder contents.

The right pane would probably be a little more of a pain than anticipated and should be saved for last if we do it.

**Comment by aviking** [comment]
Currently I have fixed the simple issue of wasted space on the left for the folder browser itself.

I have also made the adjustment that @joehof  mentioned of using the table with just the link attribute showing. I only gave the docViewer 75% of the page, because that is what it takes up in the Stage. With this I have the table taking up 25% which may seem a bit large but because we have to keep the controls showing it actaully is helpful to have a bit of extra space. 

I have made it so that if you want to see the folderbrowser still while showing the doc viewer, the docViewer takes up 50% and the other two panes take up 25%.
